PRNG.ml

```diff
@@ -143,14 +145,12 @@ let nativeint =
   then fun g bound -> Nativeint.of_int32 (int32 g (Nativeint.to_int32 bound))
   else fun g bound -> Int64.to_nativeint (int64 g (Int64.of_nativeint bound))
 
-let float_64 g bound =
+let rec uniform g =
   let b = X.bits64 g in
-  (Int64.(to_float (shift_right_logical b 11)) *. 0x1.p-53) *. bound
+  let n = Int64.shift_right_logical b 11 in
+  if n = 0L then uniform g else Int64.to_float n *. 0x1.p-53
```

Comment:
This implementation certainly works, but:
- Why masking the lowest 11 bits when we could in fact take them into account (and multiply by `0x1p-64` instead)?
- Testing `n = 0` and branching can be a bit costly. In stead, the same behaviour could be obtained by adding a small number after the multiplication (`0x1p-54` or `0x1p-65`, depending on what you choose for the first remark). Sure, then you need to change the documentation below...

Reply:
"take [all the bits] into account (and multiply by 0x1p-64 instead)": then you get FP rounding during the int64 -> float conversion.  The rounding can produce 0.0 and 1.0, and seems to introduce some (very subtle) bias in the output, as discussed in [Generating Random Floating-Point Numbers by Dividing Integers: a Case Study](https://hal.archives-ouvertes.fr/hal-02427338).  With the current code, we know exactly which FP values we can get, and we're sure they all have the same probability.

"Testing n = 0 and branching can be a bit costly".  Really?  This is one integer comparison plus a branch that is perfectly predicted (as n is almost never 0).

"adding a small number after the multiplication": the resulting rounding makes me nervous (again).

Reply:
Right, I forgot about your adventures with SIMD :-)

I take it that you find the name `uniform` acceptable for this function ?  The only alternative I could think of is `float1`, but it's ugly.